src/restoreImpl.ts

```diff
@@ -34,12 +34,13 @@ async function restoreImpl(
         const enableCrossOsArchive = utils.getInputAsBool(
             Inputs.EnableCrossOsArchive
         );
+        const dryRun = utils.getInputAsBool(Inputs.DryRun);
```

Comment:
Did you try running a sample workflow of this version of action with `cache` action? 
As cache action doesn't have this input, I'm curious if it throws any warning with this implementation?

Reply:
Not sure I understand your question correctly.

The `dry-run` input is defined for both the normal `cache` and also the `cache/restore` action with `default: 'false'`.

> I'm curious if it throws any warning with this implementation?

Even if it wouldn't be defined, the code shouldn't throw an error. `utils.getInputAsBool` uses `core.getInput` under the hood which returns an empty string for undefined variables. The result would thus still be `false`.
https://github.com/actions/cache/blob/22cbf49050c8786eaada2e90f2f3c4ea0fb20ad5/src/utils/actionUtils.ts#L55-L60

[actions/toolkit -> core.ts](https://github.com/actions/toolkit/blob/6c1f9eaae833355a0b212b66c5f2e3ac366de185/packages/core/src/core.ts#L117-L138)

Reply:
Pushed a small update for the readmes to reflect the changes in `action.yml`.

Reply:
Got it, I was under assumption that this input is only for the `restore` action.
Which brings me to my next question, how is the behaviour in case of the `cache` action? 
Usually the `cache` action saves cache in the post step when `cache-hit` output is set to `false`. Wouldn't unnecessary caches be saved when `dry-run`:`true` and no cache is found?

Reply:
> Which brings me to my next question, how is the behaviour in case of the `cache` action?
> Usually the `cache` action saves cache in the post step when `cache-hit` output is set to `false`. Wouldn't unnecessary caches be saved when `dry-run`:`true` and no cache is found?

I don't think so. The main use case I have in mind for it would depend on saving the cache afterwards. `dry-run: 'true'` is meant to speed up the case where the cache would be created from scratch anyway. If no exact cache is found, it still needs to be created and stored after that to be able to use it for the next run.

This works best, if `restore-keys` is left empty as downloading those would be skipped as well. However, it can still be archived by stacking the `cache/restore` (with `dry-run: 'true'`) and normal `cache` (with `restore-keys`) action. The first would be there to filter out exact matches, while the following steps would only run if `cache-hit` is `false`.

Reply:
Understood, in that case the term `dry-run` seems a little misleading to me if we are going to **actually save the cache** in the name of dry run in the `actions/cache` action. 😅 
Can we name it better such that the name suits both `actions/cache` and `actions/cache/restore` action?
Actual dry run is only happening in `actions/cache/restore`, `actions/cache` on the other hand still saves and this would confuse other users as only a specific set of users who are part of https://github.com/actions/cache/issues/901 issue only know the actual use case.

Reply:
> Can we name it better such that the name suits both `actions/cache` and `actions/cache/restore` action?

Sure, `dry-run` was just the first name that came to mind but I can see how it can be confusing.
What do you think of `skip-download`?

I also though about `restore-dry-run` or `skip-restore` but those don't capture the essence of this option or can be misleading as well. I'm open to other suggestions though.

Reply:
`skip-download` won't make sense for the `restore` action right? `skip-download` sounds more like `save-only`.

`restore-dry-run` is still close. My suggestion would be to use something like `check-key` (although not quite sure what the perfect name here could be), @bishal-pdMSFT do suggest us something that you feel would be good here.

Reply:
> `skip-download` won't make sense for the `restore` action right?

I'd disagree. Skipping the download is what's happing internally, regardless of `cache` or `restore` action. It's also the term I used naturally to describe the parameter in the readme. That's why I thought it might be a good fit.

> My suggestion would be to use something like `check-key`

Just thinking about it, I could imagine that users could be confused by that, especially if the default is `false`. Is the key not checked unless it's set to `true`?

Reply:
> `skip-download` won't make sense for the `restore` action right?

My reason of saying the same was - downloading is the main job of the `restore` action and the restore step in the `cache` action. For `cache` action, the name `skip-download` is still fine, but for `restore` action, if restore (download) itself is not happening then I see a chance of confusion for the users thinking what it would even do if not download the cache itself. 😕 Hence I was tilting more towards something that tell `check-key(-but-dont-download)`

Few more suggestions from my end...

How about going in the direction of `peek` like we use in stacks, just see but don't do anything. 
Examples:
`peek-key` / `peek-cache`
 or 
`lookup` / `lookup-only`
or
`spy`


>  Is the key not checked unless it's set to true?

True, it will create a confusion. 😅

Reply:
> For `cache` action, the name `skip-download` is still fine, but for `restore` action, if restore (download) itself is not happening then I see a chance of confusion for the users thinking what it would even do if not download the cache itself.

Just a thought, would it be enough to expose the option only for `cache`? On the one hand it would address the main use case, on the other there is some value in exposing the option for _all_ actions.

> Hence I was tilting more towards something that tell `check-key(-but-dont-download)`

`lookup-only` sounds like it could fit. It would make sense in the docs as well. Something like
```markdown
* `lookup-only` - Skip downloading cache. Only check if cache entry exists. Default: false
```